### PR TITLE
Add ha_resource_move action

### PR DESCRIPTION
**Pull request by utopiabound**
Move resource, wait for move completion, then remove move constraint.

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2391)

### CODE REVIEW

**Review by ip1981 — DISMISSED** [review]

**Review by johnsonw — DISMISSED** [review]

File: iml-agent/src/action_plugins/high_availability/mod.rs [hunk 1]
```diff
@@ -292,6 +622,41 @@ pub async fn stop_resource(resource: String) -> Result<(), ImlAgentError> {
...
+        let output = crm_mon_cmd().checked_output().await?;
+        let cluster = read_crm_output(&output.stdout)?;
+
+        let res = cluster
+            .resources
+            .into_iter()
+            .filter(|r| r.id == resource)
+            .next();
+        if let Some(r) = res {
+            if r.active_node_name.unwrap_or("".into()) == dest_host {
```
Comment (jgrund) [inline iml-agent/src/action_plugins/high_availability/mod.rs:641]: clippy warning.

File: iml-agent/src/action_plugins/high_availability/mod.rs [hunk 2]
```diff
@@ -292,6 +622,41 @@ pub async fn stop_resource(resource: String) -> Result<(), ImlAgentError> {
...
+
+    crm_resource(&["--resource", &resource, "--move", "--node", &dest_host]).await?;
+
+    let mut counter = 0;
+    let rc = loop {
+        let output = crm_mon_cmd().checked_output().await?;
+        let cluster = read_crm_output(&output.stdout)?;
+
+        let res = cluster
+            .resources
```
Comment (jgrund) [inline iml-agent/src/action_plugins/high_availability/mod.rs:636]: resource_list

Comment (jgrund) [reply iml-agent/src/action_plugins/high_availability/mod.rs:636]: clippy warning

File: iml-agent/src/action_plugins/high_availability/mod.rs [hunk 3]
```diff
@@ -619,6 +622,41 @@ pub async fn stop_resource(resource: String) -> Result<(), ImlAgentError> {
...
+            .into_iter()
+            .filter(|r| r.id == resource)
+            .next();
```
Comment (jgrund) [inline iml-agent/src/action_plugins/high_availability/mod.rs:639]: clippy warning.

File: iml-agent/src/action_plugins/high_availability/mod.rs [hunk 4]
```diff
@@ -619,6 +622,41 @@ pub async fn stop_resource(resource: String) -> Result<(), ImlAgentError> {
...
+            break Err(ImlAgentError::from(RequiredError(format!(
+                "Waiting for resource {} to move to {} failed after {} sec",
+                resource, dest_host, WAIT_SEC,
+            ))));
+        }
+        counter += WAIT_DELAY;
+        delay_for(delay_duration).await;
+    };
+
+    crm_resource(&["--resource", &resource, "--un-move", "--node", &dest_host]).await?;
```
Comment (jgrund) [inline iml-agent/src/action_plugins/high_availability/mod.rs:651]: `--un-move` should be called `--allow-move` or something more representative

**Review by jgrund — APPROVED** [review]

**Review by johnsonw — APPROVED** [review]

**Pull request merged**